### PR TITLE
Remove build warnings, consolidate Package versions

### PR DESCRIPTION
**Pull request by MichelZ**
Working on https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2912

- Consolidate some low-risk libraries and/or only Test projects
- Removing build warnings
- Add NSubstitute.Analyzers (as per NSub Docs best practices)
  - Note: This adds some NSub specific warnings for things that don't seem to be done "correctly"

### CODE REVIEW

File: tests/Microsoft.Identity.Test.Integration.netcore/Microsoft.Identity.Test.Integration.NetCore.csproj [hunk 1]
```diff
@@ -33,6 +33,7 @@
     <PackageReference Include="StrongNamer">
       <Version>0.2.5</Version>
     </PackageReference>
+    <PackageReference Include="System.Security.Cryptography.Cng" Version="5.0.0" />
```
Comment (bgavrilMS) [inline tests/Microsoft.Identity.Test.Integration.netcore/Microsoft.Identity.Test.Integration.NetCore.csproj:36]: Why is this needed?

File: tests/Microsoft.Identity.Test.Performance/Microsoft.Identity.Test.Performance.csproj [hunk 2]
```diff
@@ -27,6 +27,7 @@
   <ItemGroup>
     <PackageReference Include="BenchmarkDotNet" Version="0.13.1" />
     <PackageReference Include="BenchmarkDotNet.Diagnostics.Windows" Version="0.13.1" />
+    <PackageReference Include="System.CodeDom" Version="6.0.0" />
```
Comment (bgavrilMS) [inline tests/Microsoft.Identity.Test.Performance/Microsoft.Identity.Test.Performance.csproj:30]: Why is this needed?

File: tests/Microsoft.Identity.Test.Integration.netcore/Microsoft.Identity.Test.Integration.NetCore.csproj (hunk 1 above)
Comment (MichelZ) [reply tests/Microsoft.Identity.Test.Integration.netcore/Microsoft.Identity.Test.Integration.NetCore.csproj:36]: Resolving this warning
> Warning	MSB3277	Found conflicts between different versions of "System.Security.Cryptography.Cng" that could not be resolved.
> There was a conflict between "System.Security.Cryptography.Cng, Version=4.3.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" and "System.Security.Cryptography.Cng, Version=4.3.3.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a".
>     "System.Security.Cryptography.Cng, Version=4.3.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" was chosen because it was primary and "System.Security.Cryptography.Cng, Version=4.3.3.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" was not.
>

**Review by bgavrilMS — APPROVED** [review]

**Review by pmaytak — APPROVED** [review]
